packages/block-editor/src/store/private-selectors.js

```diff
@@ -300,10 +300,10 @@ export const hasAllowedPatterns = createRegistrySelector( ( select ) =>
 				if ( ! inserter ) {
 					return false;
 				}
-				const { blocks } = getParsedPattern( pattern );
+				const grammar = getGrammar( pattern );
```

Comment:
Grammar parse is lightening fast, but probably still a good idea to cache.

Reply:
It's worth measuring the impact if possible, sometimes caching adds more overhead than the actual operation.